tests/src/test/scala/org/http4s/UriSpec.scala

```diff
@@ -627,6 +627,16 @@ http://example.org/a file
       val u = Uri() +?? ("param1", None: Option[Int])
       u must be_==(Uri(query = Query.empty))
     }
+    "add multiple query parameters at once" in {
+      val params = Map("param1" -> 1, "param2" -> 2)
+      val u = Uri().withQueryParams(params)
+      u must be_==(Uri(query = Query.fromString("param1=1&param2=2")))
+    }
+    "add multiple values for same query parameter name" in {
+      val params = Map("param1" -> List(1), "param2" -> List(2, 3))
+      val u = Uri().withMultiValueQueryParams(params)
+      u must be_==(Uri(query = Query.fromString("param1=1&param2=2&param2=3")))
```

Comment:
Looks like this is the way http4s handles multi-value params. 
Another convention seems to be `param1=1&param2=2,3`.

Reply:
The comma as a separator is arbitrary.  What if your data has a comma?  There's no single way to do it.  There's also convention in some dynamically typed languages to flag multi-values with an array syntax: `param[]=1,param[]=2`.  None of these conventions are precluded by http4s, but the repeated key has a fairly straightforward interpretation.